src/platform/plugins/shared/workflows_management/server/workflows_management/workflows_management_service.ts

```diff
@@ -100,12 +108,39 @@ export class WorkflowsService {
   }
 
   public async searchWorkflows(params: GetWorkflowsParams): Promise<WorkflowListDto> {
+    const { limit, page, query, createdBy, status } = params;
+
     const savedObjectsClient = await this.getSavedObjectsClient();
+
+    const filters: string[] = [];
+
+    if (createdBy && createdBy.length > 0) {
+      const createdByFilter = createdBy
+        .map((user) => `${SO_ATTRIBUTES_PREFIX}.createdBy:"${user}"`)
+        .join(' OR ');
+      filters.push(`(${createdByFilter})`);
+    }
+
+    if (status && status.length > 0) {
+      const statusFilter = status.map((s) => `${SO_ATTRIBUTES_PREFIX}.status:"${s}"`).join(' OR ');
+      filters.push(`(${statusFilter})`);
+    }
+
+    if (query) {
+      filters.push(
+        `(${SO_ATTRIBUTES_PREFIX}.name:"${query}" OR ${SO_ATTRIBUTES_PREFIX}.description:"${query}")`
+      );
+    }
+
+    const filterQuery = filters.join(' AND ');
+
     const response = await savedObjectsClient.find<WorkflowSavedObjectAttributes>({
```

Comment:
Noticed a few bugs:
1. pagination seems to work incorrectly: changing page doesn't change results 
2. partial name in search doesn't return, e.g. ("slac" will not return workflows with names "slack 1", "slack 2")

Video:

https://github.com/user-attachments/assets/4b3cbb9a-bed4-4e43-86e1-54cb8dd89511